web/source/kmwlayout.js

```diff
@@ -183,8 +183,12 @@ if (!window['tavultesoft']['keymanweb']['initialized']) {
                                 {
                                     "id": "K_SHIFT", "text": "*Shift*", "sp": "1", "width": "200", "sk": [
                                         { "id": "K_LCONTROL", "text": "*Ctrl*", "sp": "1", "width": "50", "nextlayer": "ctrl" },
+                                        { "id": "K_LCONTROL", "text": "*LCtrl*", "sp": "1", "width": "50", "nextlayer": "leftctrl" },
+                                        { "id": "K_RCONTROL", "text": "*RCtrl*", "sp": "1", "width": "50", "nextlayer": "rightctrl" },
                                         { "id": "K_LALT", "text": "*Alt*", "sp": "1", "width": "50", "nextlayer": "alt" },
-                                        { "id": "K_ALTGR", "text": "*AltGr*", "sp": "1", "width": "50", "nextlayer": "ctrlalt" }]
+                                        { "id": "K_LALT", "text": "*LAlt*", "sp": "1", "width": "50", "nextlayer": "leftalt" },
+                                        { "id": "K_RALT", "text": "*RAlt*", "sp": "1", "width": "50", "nextlayer": "rightalt" },
+                                        { "id": "K_ALTGR", "text": "*AltGr*", "sp": "1", "width": "50", "nextlayer": "ctrl-alt" }]
```

Comment:
`K_ALTGR` is not a valid identifier; I can see what you are wanting to do here but we'll need to find another way to specify that key (can it be implemented by looking for both alt and control selected?)

Reply:
... then our mobile layouts were never valid in the first place?  I merely corrected it; it already existed.

Reply:
Keep in mind these two are for the mobile layouts 'phone' and 'tablet' - neither of which has any actual modifiers or physical keyboard related to them.

Reply:
Okay, let's discuss this in the office -- I'm not sure what the history behind the use of `K_ALTGR` in the code is here but it's not a standard Keyman virtual key identifier so we need to be careful with overloads.

Reply:
If `K_ALTGR` ends up staying, can we keep the nextlayer name "ctrlalt"? For consistency, none of the other layers have hyphenated names.

Reply:
Part of the chirality design doc (as per @mcdurdin's request) is that we use kebab-case when combining modifiers to obtain a layer name.  `ctrl-alt` is the only nextlayer comprised of multiple modifiers here, hence why it's the only one with a hyphen.

Without the hyphen, the nextlayer operation will break without somewhat significant changes to the layer id management scheme in this PR.

Reply:
Just making sure here: existing keyboards don't have kebab-cased layer names. We need to ensure this doesn't break existing keyboards.

Reply:
Hmm.  Existing keyboards that would check for this wouldn't be using the `buildDefaultLayout` method though, would they?

There _is_ the matter of the default layer-management functions, though, and if a keyboard deliberately chose the name `ctrlalt` to use said functions this may prove an issue.  Know of any good keyboard examples I could use for testing this - ones that provide their own visual layout but prefer to rely on automated layer management?

Then again, that doesn't make sense does it?  Touch layouts shouldn't respond to physical keystrokes, so there shouldn't be any automated layer management to speak of here.  Desktop devices don't provide their own OSK layout data, thus shouldn't be invoking specific layer names, right?

Reply:
A keyboard can target both desktop and touch though. I'm not sure I understand the question: how can a keyboard provide its own visual layout (do you mean .kvk visual keyboard or touch layout?) and also rely on automated layer management?
 https://github.com/keymanapp/keyboards/blob/master/fv_xaislakala_kmw/source/fv_xaislakala_kmw.kmn is a sample keyboard that has both touch and ctrl+alt.